### PR TITLE
cpu: aarch64: Expand brgemm aarch64 unsupported cases handling mechanism

### PR DESCRIPTION
**Pull request by Radu2k**
Remove the asserts that cause segfaults in the test and handle unsupported cases based on oneDNN status type. As well remove the forgotten asserts in brgemm_matmul.cpp and modify acl_deconvolution.hpp to make op/f32/deconv_bwd_d test pass.

# Description

This replaces the asserts in brgemm that make `test_benchdnn_modeC_graph_ci_cpu` segfault on AArch64 by replacing the asserts with `status_t` type in oneDNN. As well, the segfault of `--graph --skip-impl=ref --case=op/f32/deconv.json` test when built with gcc-10 and g++-10, which is part of the graph test batch.

# Checklist

## General

- [X] Do all unit and benchdnn tests (`make test` and `make test_benchdnn_*`) pass locally for each commit?
- [X] Have you formatted the code using clang-format?

## Bug fixes

- [X] Have you included information on how to reproduce the issue (either in a github issue or in this PR)?
- [X] Have you added relevant regression tests?

### CODE REVIEW

File: tests/benchdnn/graph/ref_primitive.cpp
Comment (dzarukin) [inline tests/benchdnn/graph/ref_primitive.cpp:249]: Why is this change needed? What's the issue with `auto`?

Comment (Radu2k) [reply tests/benchdnn/graph/ref_primitive.cpp:249]: @dzarukin Hi Dmitry, thanks for having a look into this. I have replaced auto with explicit data types for improved readability, consistency, and type safety. When debugging it was particularly useful to have the data type specified and I did go an extra mile checking that the function returns the same type. Should not be a problem but I am happy to review it if it is a deal breaker.

Comment (dzarukin) [reply tests/benchdnn/graph/ref_primitive.cpp:249]: It's fine with me, just wanted to be sure there's no compiler/code issue involved. Generally, all `dnn_mem_t` are either named as mem_XXX or m_XXX or plain mem/m if that helps for future debugging.

Happy to approve. BTW, QQ: do you have Graph API working on aarch64?

Comment (Radu2k) [reply tests/benchdnn/graph/ref_primitive.cpp:249]: I’ve been working on this the past few days, and all tests for the Graph API should run fine. There are a few edge cases that fail but I’ll detail in an issue soon and shall be fixed in the foreseeable future. 

Thanks for the approval!

**Review by mgouicem — APPROVED** [review]
Could you split the change to acl_deconvolution in a separate commit as it seems to fix an issue?

**Comment by theComputeKid** [comment]
@mgouicem and @dzarukin can you review on behalf of the graph team and merge?

**Comment by mgouicem** [comment]
We are not approvers for graph component. @TaoLv could you help?

**Comment by vpirogov** [comment]
The change that triggers review from Graph component is discussed [here](https://github.com/oneapi-src/oneDNN/pull/2099#discussion_r1765525498). Considering that it's purely stylistical it may make sense to just drop it.

**Comment by Radu2k** [comment]
> The change that triggers review from Graph component is discussed [here](https://github.com/oneapi-src/oneDNN/pull/2099#discussion_r1765525498). Considering that it's purely stylistical it may make sense to just drop it.

Just to clarify on this topic. This is needed due to the segfaulting behaviour of the test. These are on **high** priority because segfaults are a security risk. The patch that has introduced this is not coming from us and we will take further action once this change has been submitted. We already have an Issue prepared. 

We are promptly on a clear path to minimise our tolerance for any tests that do not guard properly any unsupported cases or any PRs that have slipped trough our fingers in the past.

**Comment by theComputeKid** [comment]
@vpirogov and @mgouicem : removed the graph-related change, should be good to merge.

**Comment by Shreyas-fuj** [comment]
@mgouicem , @Radu2k after this PR is merged the matmul calls are going to gemm:jit:f32, instead of going to brgemm:sve_256. This is because of an incorrect conversion of **assert** to **if** statement in /src/cpu/aarch64/brgemm/brgemm_utils.cpp .

**Comment by theComputeKid** [comment]
Thanks for letting us know. @Radu2k can you please take a look?

**Comment by theComputeKid** [comment]
@Shreyas-fuj : I can see that the issue is that the conditions of the assert were flipped:
```
    if (dt_a != data_type::undef && dt_b != data_type::undef)
         return status::unimplemented;
```

```
assert(dt_a != data_type::undef && dt_b != data_type::undef);
```

These two are not equivalent, but flipped. I'll fix. Thanks once again for picking this up.

**Comment by Ryo-not-rio** [comment]
@Shreyas-fuj Thank you for bringing the issue to our attention. [This PR](https://github.com/oneapi-src/oneDNN/pull/2148) is the fix for the observed behaviour

**Comment by Shreyas-fuj** [comment]
@theComputeKid , @Ryo-not-rio , thanks for the fix!

**Pull request merged**
